src/platform_impl/redox/mod.rs

```diff
@@ -0,0 +1,1336 @@
+                            );
+
+                            // Acknowledge resize after event loop
+                            resize_opt = Some((width, height));
+                        }
+                        //TODO: Clipboard
+                        EventOption::Hover(HoverEvent { entered }) => {
+                            if entered {
+                                event_handler(
+                                    event::Event::WindowEvent {
+                                        window_id: window::WindowId(window_id),
+                                        event: event::WindowEvent::CursorEntered {
+                                            device_id: event::DeviceId(DeviceId),
+                                        },
+                                    },
+                                    &self.window_target,
+                                    &mut control_flow,
+                                );
+                            } else {
+                                event_handler(
+                                    event::Event::WindowEvent {
+                                        window_id: window::WindowId(window_id),
+                                        event: event::WindowEvent::CursorLeft {
+                                            device_id: event::DeviceId(DeviceId),
+                                        },
+                                    },
+                                    &self.window_target,
+                                    &mut control_flow,
+                                );
+                            }
+                        }
+                        other => {
+                            warn!("unhandled event: {:?}", other);
+                        }
+                    }
+                }
+
+                if count == event_buf.len() {
+                    // If event buf was full, process same window again to ensure all events are drained
+                    continue;
+                }
+
+                // Acknowledge the latest resize event
+                if let Some((w, h)) = resize_opt.take() {
+                    window
+                        .write(format!("S,{},{}", w, h).as_bytes())
+                        .expect("failed to acknowledge resize");
+
+                    // Require redraw after resize
+                    let mut redraws = self.window_target.p.redraws.lock().unwrap();
+                    if !redraws.contains(&window_id) {
+                        redraws.push_back(window_id);
+                    }
+                }
+
+                // Move to next window
+                i += 1;
+            }
+
+            while let Ok(event) = self.window_target.p.user_events_receiver.try_recv() {
+                event_handler(
+                    event::Event::UserEvent(event),
+                    &self.window_target,
+                    &mut control_flow,
+                );
+            }
```

Comment:
I think we deliver User events before other window events, but I'm not sure if it's stated anywhere.

Reply:
It's not really defined, see the top of `event.rs`.

On macOS, device and window events are delivered before user events

Reply:
I'm marking this as resolved, if macOS uses the same order